### PR TITLE
[quant] Fix qmin/qmax when using customized qrange

### PR DESCRIPTION
**Pull request by terrychenism**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #74717

Summary:
currently the weight map to 0 and max_float to 65535 due to incorrect qmin/qmax in qin16 customized qrange
the expectation from the set observers is the integer representation is supposed to be a signed int16 i.e -32768 to 32767.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D35129924](https://our.internmc.facebook.com/intern/diff/D35129924)

### CODE REVIEW

**Comment by terrychenism** [comment]
@terrychenism has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35129924).

**Comment by terrychenism** [comment]
@terrychenism has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35129924).

**Review by jerryzh168 — APPROVED** [review]
thanks! please make sure ci passes

**Comment by terrychenism** [comment]
@terrychenism has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35129924).

File: torch/ao/quantization/utils.py [hunk 1]
```diff
@@ -326,10 +326,6 @@ def calculate_qmin_qmax(quant_min: int, quant_max: int, has_customized_qrange: b
             assert (
                 0 < qrange_len <= 2**31
             ), "quantization range should be positive and not exceed the maximum bit range (=4294967296)."
-        if dtype == torch.qint8:
-            quant_min, quant_max = -qrange_len // 2, qrange_len // 2 - 1
-        else:
-            quant_min, quant_max = 0, qrange_len - 1
```
Comment (jerryzh168) [inline torch/ao/quantization/utils.py:332]: I think you can remove the next line as well

Comment (terrychenism) [reply torch/ao/quantization/utils.py:332]: do you mean remove 'reduce_range' if statement?

Comment (jerryzh168) [reply torch/ao/quantization/utils.py:332]: yes, L333-L334

Comment (terrychenism) [reply torch/ao/quantization/utils.py:332]: if we remove it, how to handle the qmin/max if user wanna use has_customized_qrange with reduce_range flag?

Comment (terrychenism) [reply torch/ao/quantization/utils.py:332]: it would break several test cases if we remove it.

**Comment by terrychenism** [comment]
@terrychenism has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35129924).

**Comment by terrychenism** [comment]
@terrychenism has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35129924).

**Comment by terrychenism** [comment]
@terrychenism has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35129924).

**Comment by terrychenism** [comment]
@terrychenism has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35129924).

File: torch/ao/quantization/utils.py (hunk 1 above)
Comment (jerryzh168) [reply torch/ao/quantization/utils.py:332]: user should set the range correctly in the beginning directly I think?

**Pull request closed without merging**